doc/read-the-docs-site/adr/0004-deferred-unlifting.md

```diff
@@ -0,0 +1,78 @@
+# ADR 4: Deferred unlifting in Plutus Core
+
+Date: 2022-11
+
+## Authors
+
+Michael Peyton Jones <michael.peyton-jones@iohk.io>
+
+## Status
+
+Accepted
+
+## Context
+
+A key part of the evaluation of builtin applications in Plutus Core is "unlifting".
+Unlifting is the process of taking a Plutus Core term and turning it into a Haskell value of a known type.
+For example, we can unlift an integer constant term into the actual Haskell integer it contains.
+This is necessary in order to apply the denotation of the builtin being applied, since that is a Haskell function that operates on Haskell types (e.g. integer addition).
+
+However, unlifting can fail: we cannot unlift a _string_ constant into a Haskell integer!
+This failure is visible in program execution, since it terminates the program with an error.
+
+The original design of the builtin application machinery performed unlifting of an argument as soon as it was received. 
+This meant that unlifting failures would surface at that point, whereas most of the errors that relate to builtin evaluation can only occcur once the builtin has all its arguments, since that's when we run the actual function.
```

Comment:
```suggestion
This meant that unlifting failures would surface at that point, whereas most of the errors that relate to builtin evaluation can only occur once the builtin has all its arguments, since that's when we run the actual function.
```